pkg/asset/installconfig/azure/validation.go

```diff
@@ -158,6 +158,29 @@ func validateMininumRequirements(fieldPath *field.Path, req resourceRequirements
 	return allErrs
 }
 
+func validateSecurityType(fieldPath *field.Path, securityType aztypes.SecurityTypes, instanceType string, capabilities map[string]string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	unsupportedSecurityType := false
+	_, hasTrustedLaunchDisabled := capabilities["TrustedLaunchDisabled"]
+	confidentialComputingType, hasConfidentialComputingType := capabilities["ConfidentialComputingType"]
+	isConfidentialComputingTypeSNP := confidentialComputingType == "SNP"
+
+	switch securityType {
+	case aztypes.SecurityTypesConfidentialVM:
```

Comment:
Nit: Is it at all possible to make this code represent cases that are supported and the default represent the unsupported cases?

Reply:
Good point, I also think it's more readable this way and updated this code accordingly PTAL